### PR TITLE
KOGITO-6787: Serverless Workflow validation

### PR DESCRIPTION
**Pull request by Sara4994**
Hello,

Please review this pull requesting which adds validation errors to the VSCode Problems Tab.

JIRA link: https://issues.redhat.com/browse/KOGITO-6787

### CODE REVIEW

**Comment by domhanak** [comment]
is there any JIRA for this?

**Comment by domhanak** [comment]
please make sure you tag ask for QE review for everything SWF tooling related. Thanks

**Comment by Sara4994** [comment]
> please make sure you tag ask for QE review for everything SWF tooling related. Thanks

@domhanak Sure, will make sure I do that next time. Thanks :+1:

File: packages/notifications/src/vscode/ProblemsTabNotificationHandler.ts [hunk 1]
```diff
@@ -52,9 +52,12 @@ export class ProblemsTabNotificationHandler implements NotificationsApi {
   }
 
   private buildDiagnostic(notification: Notification): vscode.Diagnostic {
+    const startLineNumber = notification.position?.startLineNumber ? notification.position?.startLineNumber - 1 : 0;
+    const startColumn = notification.position?.startColumn ? notification.position?.startColumn - 1 : 0;
+    const endColumn = notification.position?.endColumn || 0;
     return {
       message: notification.message,
-      range: new vscode.Range(new vscode.Position(0, 0), new vscode.Position(0, 0)),
+      range: new vscode.Range(startLineNumber, startColumn, startLineNumber, endColumn),
```
Comment (paulovmr) [inline packages/notifications/src/vscode/ProblemsTabNotificationHandler.ts]: Should it be `endLineNumber` at the third parameter?

**Review by paulovmr — CHANGES_REQUESTED** [review]

**Comment by barboras7** [comment]
Hi @Sara4994,
I tried to test the validation but I did not succeed. Here's what I did on the swf-validation branch:

- `yarn build:prod:until @kie-tools-core/notifications`
- `yarn build:prod:until vscode-extension-serverless-workflow-editor`

I took the `vscode_extension_serverless_workflow_editor_0.0.0.vsix` and installed it on VSCode. I tried testing with both yaml and json. I took a working *.sw.json and *.sw.yaml. I edited the files in the VSCode so that they were no longer valid. Is this how the feature should be tested? Because I did not see anything in the Problems tab.

**Review by paulovmr — CHANGES_REQUESTED** [review]
@Sara4994 The code looks good to me, only thing is that sometimes the problems are duplicated in the problems tab:
image

**Review by paulovmr — APPROVED** [review]

**Comment by paulovmr** [comment]
The previous issue should be fixed now.

@barboras7 Could you try testing it again please? Thanks.

**Comment by tiagobento** [comment]
@Sara4994 There are some conflicts here.. can you please address them?

**Review by barboras7 — APPROVED** [review]
The PR looks good, thanks @Sara4994 :+1:

**Comment by Sara4994** [comment]
> @Sara4994 There are some conflicts here.. can you please address them?

@tiagobento Conflicts are fixed :+1

**Comment by tiagobento** [comment]
@Sara4994 Please check the formatting of the files.

**Comment by paulovmr** [comment]
I've added the form-dmn dependency to unitables-dmn.

**Pull request merged**